tests/ptf/fabric.ptf/test.py

```diff
@@ -262,6 +262,38 @@ def runTest(self):
         testutils.send_packet(self, self.port2, str(pkt_2to1))
         testutils.verify_packets(self, exp_pkt_2to1, [self.port1])
 
+class FabricL2UnicastVlanTest(FabricTest):
+    @autocleanup
+    def runTest(self):
+        mac_addr_mask = ":".join(["ff"] * 6)
+        vlan_id = 10
+        # set internal VLAN for port 2 only since packet from port 1 is tagged
+        self.set_internal_vlan(self.port2, False, 0, 0, vlan_id)
+        # miss on filtering.fwd_classifier => bridging
+        self.add_bridging_entry(vlan_id, HOST1_MAC, mac_addr_mask, 10)
+        self.add_bridging_entry(vlan_id, HOST2_MAC, mac_addr_mask, 20)
+        self.add_next_hop(10, self.port1)
+        self.add_next_hop(20, self.port2)
+        # pops VLAN on port 2 since port 2 is an untagged port
+        self.set_egress_vlan_pop(self.port2, vlan_id)
+
+        pkt_1to2 = testutils.simple_tcp_packet(
+            eth_src=HOST1_MAC, eth_dst=HOST2_MAC, dl_vlan_enable=True, vlan_vid=vlan_id, ip_ttl=64)
+        exp_pkt_1to2 = testutils.simple_tcp_packet(
+            eth_src=HOST1_MAC, eth_dst=HOST2_MAC,
+            ip_ttl=63, pktlen=96) # packet length will decrease
```

Comment:
Why do you need to specify `pktlen` here? I might be wrong, but if the VLAN header is removed from `pkt_1to2`, I expect its length to be the same as a packet returned by `testutils.simple_tcp_packet()` without setting a VLAN ID.

Reply:
the simple_tcp_packet generates payload according to pktlen(default is 100) and total length of headers

So the payload length of `pkt_1to2` will be 41, and `exp_pkt_1to2` will be 45 if we don't change the pktlen

Reply:
Got it, thanks!